controller/k8s/clientset.go

```diff
@@ -1,14 +1,15 @@
 package k8s
 
 import (
+	spclient "github.com/linkerd/linkerd2/controller/gen/client/clientset/versioned"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	// Load all the auth plugins for the cloud providers.
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 )
 
-func NewClientSet(kubeConfig string) (*kubernetes.Clientset, error) {
+func NewClientSet(kubeConfig string) (*kubernetes.Clientset, *spclient.Clientset, error) {
```

Comment:
Since most processes that need a kubernetes clientset are not also going to need an sp clientset, I'd prefer to have 2 separate methods in this file: `NewClientSet` and `NewSpClientSet`. You could create a separate helpful func for config parsing that's shared between the two methods.